syntax = "proto3";

service Whisper {
    rpc LocalTranscribeAnimeDub (LocalTranscribeAnimeDubRequest) returns (LocalTranscribeAnimeDubResponse) {}
}
// set enum for model of local transcribe request: tiny.en 

message LocalTranscribeAnimeDubRequest {
    string path = 1;
    Model model = 2;
}

message LocalTranscribeAnimeDubResponse {
    string text = 1;
}

enum Model {
    TINY = 0;
    BASE = 1;
    SMALL = 2;
    MEDIUM = 3;
    LARGE = 4;
    TINY_EN = 5;
    BASE_EN = 6;
    SMALL_EN = 7;
    MEDIUM_EN = 8;
  }